### PR TITLE
Exports metrics to track non primary container resource requests

### PR DESCRIPTION
**Pull request by jiayingz**
**What this PR does**
- Exports metrics on container non-primary resource requests.
- Adds "kubernetes.io/container/nvidia_gpu/request" metric generated from such metrics to stackdriver sink.

### CODE REVIEW

File: metrics/processors/pod_based_enricher.go [hunk 1]
```diff
@@ -182,6 +183,31 @@ func updateContainerResourcesAndLimits(metricSet *core.MetricSet, container kube
 		metricSet.MetricValues[core.MetricMemoryRequest.Name] = intValue(0)
 	}
 
+	for key, val := range container.Resources.Requests {
+		if key == kube_api.ResourceCPU || key == kube_api.ResourceMemory {
```
Comment (jiayingz) [inline metrics/processors/pod_based_enricher.go]: @jingxu97 once you add the metric for local storage, I think you want to add it here as well.

File: metrics/core/metrics.go [hunk 2]
```diff
@@ -141,6 +141,14 @@ var NetworkMetrics = []Metric{
 	MetricNetworkTxRate,
 }
 
+// Maps from non primary resource name (i.e., not cpu or memory) to the metric
+// that tracks container resource request on that resource. The name of the metric
+// is in format of formattedResourceName/request where formattedExtendedResourceName
+// is generated from the resource name with "/" replaced by "_".
+// E.g., a container requesting nvidia.com/gpu resource will generates a
+// nvidia.com_gpu/request metric in this map.
+var NonPrimaryResourceRequestMetrics = map[string]Metric{}
```
Comment (kawych) [inline metrics/core/metrics.go]: Can you put all resource request metrics here, both primary and extended? You can initialize a map with existing primary resource metric. I don't think the logic should be different for extended resources.

What about limits?

File: metrics/processors/pod_based_enricher.go (hunk 1 above)
Comment (kawych) [reply metrics/processors/pod_based_enricher.go]: Note that we have kube_api.ResourceEphemeralStorage.

**Review by jiayingz — COMMENTED** [review]
Thanks a lot for the comments!

File: metrics/sinks/stackdriver/stackdriver.go [hunk 3]
```diff
@@ -613,6 +613,9 @@ func (sink *StackdriverSink) TranslateMetric(timestamp time.Time, labels map[str
 				"memory_type": "evictable",
 			}
 			return ts
+		case "nvidia.com/gpu/request":
```
Comment (kawych) [inline metrics/sinks/stackdriver/stackdriver.go]: Keep in mind that this exposes the metrics for customers using new resource model. New resource model is going to be opt-in in GKE for Kubernetes 1.10 and legacy resource modes is still going to be a default. If you want to expose this metric also to customers with new resource model, please add a second (legacy metric). The appropriate name would be "container.googleapis.com/container/nvidia_gpu/request" and it should be reported against legacy "gke_container" resource type.

**Comment by kawych** [comment]
LGTM once the metric is added to Stackdriver. Please take a look at presubmits and consider adding a corresponding metrics for legacy resource model. Thanks.

**Comment by jiayingz** [comment]
@kawych I think this PR is ready for review now. Thanks!

**Comment by jiayingz** [comment]
/cc @jingxu97 @andyxning 
ping @kawych

**Pull request merged**